librocksdb_sys/crocksdb/c.cc

```diff
@@ -3331,6 +3331,14 @@ void crocksdb_lru_cache_options_set_high_pri_pool_ratio(
   opt->rep.high_pri_pool_ratio = high_pri_pool_ratio;
 }
 
+void crocksdb_lru_cache_options_set_no_dump_allocator(
```

Comment:
To make the API more general, we should
* add a `crocksdb_memory_allocator_t` to wrap around `rocksdb::MemoryAllocator`
* add a `crocksdb_create_jemalloc_nodump_allocator()` to call `NewJemallocNodumpAllocator`, and
* add a `crocksdb_lru_cache_options_set_memory_allocator()` to set the allocator.

We may later want to extent the allocator, or implement other types of allocator.

Same on rust side (src/rocksdb_options.rs or src/rocksdb.rs), we should implement a `MemoryAllocator` type which wraps around `crocksdb_memory_allocator_t`.

Reply:
for the `MemoryAllocator` rust wrapper, we don't need to implement any method, except for a `new_jemalloc_nodump_allocator()` method which create an instance, which can be set to the `LRUCacheOptions` rust wrapper.

Reply:
@yiwu-arbug thank you, please take a look again?